### PR TITLE
Allow to configure cgroupsv1 per nodepool

### PR DESCRIPTION
**Pull request by fiunchinho**
### What does this PR do?

Towards https://github.com/giantswarm/roadmap/issues/3676

### What is the effect of this change to users?

Users may configure different node pools with different cgroups configuration.
Also, control plane nodes always use cgroups v2.

### Any background context you can provide?

Before this change, the configuration applied to all node pools and the control plane.

I could not find a good way of using the same `config.toml` template to render differently for the control plane nodes and the node pools. For some reason, Helm doesn't allow me to pass a custom `dict` of parameters when using the `tpl` function to render the `config.toml` file. I can't access subfields from the templat ewhen I do that, only fields at the root level. I tried different approaches, even asked in Slack, and we couldn't figure it out. So I ended up I following the principle of least astonishment having two different `config.toml` files: one for the control plane nodes, and another one for the worker nodes. It's not great but easy to understand. When customers finish migrating to cgroups v2, we don't need different configuration between node pools and control plane nodes, so at that moment we could remove this hack and keep only one config.toml file.

### Should this change be mentioned in the release notes?

- [X] CHANGELOG.md has been updated (if it exists)

### CODE REVIEW

File: helm/cluster/templates/clusterapi/_helpers_files.tpl
Comment (Gacko) [inline helm/cluster/templates/clusterapi/_helpers_files.tpl:40]: Where is `$.nodePool` coming from here?

File: helm/cluster/values.schema.json
Comment (Gacko) [inline helm/cluster/values.schema.json:761]: Please don't tell me it's a capital C because some validation requires this... 🙄🤣

File: helm/cluster/values.schema.json
Comment (Gacko) [inline helm/cluster/values.schema.json:1699]:
```suggestion
                            "description": "Force use of CGroups v1 for whole cluster. Deprecated: Use the node pool level setting instead.",
```

File: helm/cluster/values.schema.json
Comment (Gacko) [reply helm/cluster/values.schema.json:761]: Also: Would it make sense, to have this as a string? Just in case something like a v3 is coming one day.

File: CHANGELOG.md
Comment (Gacko) [inline CHANGELOG.md:13]: Why a new line here? Wouldn't it make more sense to have it in the previous line?

File: helm/cluster/templates/clusterapi/_helpers_files.tpl
Comment (AndiDog) [reply helm/cluster/templates/clusterapi/_helpers_files.tpl:38]:
```
{{- $_ := set $ "nodePool" (dict "name" $nodePoolName "config" $nodePoolConfig) }}
```

File: helm/cluster/values.schema.json
Comment (fiunchinho) [reply helm/cluster/values.schema.json:761]: It's not like we just put the value of the variable in the configuration, but depending on the boolean value we run some logic to render different things. So I don't think we get much value out of making it a string, If a v3 comes along, we don't know what changes that may require.

File: helm/cluster/templates/clusterapi/_helpers_files.tpl
Comment (fiunchinho) [reply helm/cluster/templates/clusterapi/_helpers_files.tpl:38]: If you start to follow the actual chain of function calls, templates rendering and so on, you could start forgetting who you are, but also [find it here](https://github.com/giantswarm/cluster/blob/main/helm/cluster/templates/clusterapi/workers/kubeadmconfig.yaml#L16).

File: helm/cluster/values.schema.json
Comment (Gacko) [reply helm/cluster/values.schema.json]: Sure, sure, I was just thinking of some kind of "switch case" logic on the string then. That would make it more flexible than having to change all the code instead of just adding another `case`.

**Comment by fiunchinho** [comment]
[CAPA e2e tests passing](https://github.com/giantswarm/cluster-aws/pull/866#issuecomment-2382593404)

File: helm/cluster/files/etc/containerd/controlplane-config.toml
Comment (fiunchinho) [inline helm/cluster/files/etc/containerd/controlplane-config.toml:22]: Always use cgroups v2 in the control plane.

File: helm/cluster/templates/clusterapi/_helpers_files.tpl
Comment (fiunchinho) [inline helm/cluster/templates/clusterapi/_helpers_files.tpl:5]: All the removed lines in this file were moved to the workers helpers file , because now it's only used for worker nodes.

File: helm/cluster/README.md
Comment (fiunchinho) [reply helm/cluster/README.md]: Right! Thanks

File: helm/cluster/templates/clusterapi/workers/_helpers_files.tpl
Comment (fiunchinho) [reply helm/cluster/templates/clusterapi/workers/_helpers_files.tpl]: Good catch! thanks.

File: helm/cluster/files/etc/containerd/workers-config.toml
Comment (fiunchinho) [reply helm/cluster/files/etc/containerd/workers-config.toml:22]: Good catch! I wanted to use while the field is deprecated. We can removed when we remove the field.

**Comment by fiunchinho** [comment]
I'm currently blocked because e2e tests won't use the right version https://gigantic.slack.com/archives/C07KSM2E51A/p1727967655453349

**Review by AndiDog — APPROVED** [review]

**Comment by Gacko** [comment]
Tests in `cluster-aws` are working again and are actually green. Feel free to merge!

**Review by AndiDog — APPROVED** [review]

**Comment by fiunchinho** [comment]
@Gacko I need approval from a code owner. Could you approve?

**Review by Gacko — APPROVED** [review]
Oh, sorry! I forgot about this... so sorry! Here's your approval, I'm really happy to see this merged! 🥳

**Pull request merged**
